src/ray/raylet/node_manager.cc

```diff
@@ -562,6 +566,15 @@ void NodeManager::ProcessClientMessage(
     // because it's already disconnected.
     return;
   } break;
+  case protocol::MessageType::IntentionalDisconnectClient: {
+    if (registered_worker) {
+      registered_worker->MarkDead();
```

Comment:
I think this can be removed. Do we need it?

`MarkDead` I think is only used when a driver dies to essentially ignore messages from workers that were executing tasks for that driver.

However, by `returning in this method and not calling `client->ProcessMessages();`, I think that will happen automatically.

Reply:
Well, the point of this is to avoid the warning message here:

https://github.com/ray-project/ray/blob/9c14050e8b20a3372f8039268adb587eec05a887/src/ray/raylet/node_manager.cc#L678-L685

that is raised in this code path (but avoided if we set it as `dead`): https://github.com/ray-project/ray/blob/9c14050e8b20a3372f8039268adb587eec05a887/src/ray/raylet/node_manager.cc#L662-L664

Is there a better way?

Reply:
Oh hm, can we just add a `bool push_warning` argument to `ProcessDisconnectClientMessage`?